src/ucp/tag/rndv.c

```diff
@@ -26,7 +27,10 @@ size_t ucp_tag_rndv_rts_pack(void *dest, void *arg)
     rndv_rts_hdr->size             = sreq->send.length;
 
     /* Pack remote keys (which can be empty list) */
-    if (UCP_DT_IS_CONTIG(sreq->send.datatype)) {
+    if (UCP_DT_IS_CONTIG(sreq->send.datatype) &&
```

Comment:
this is duplicate code, need to add a small function `ucp_rndv_is_get_zcopy(sreq, rndv_mode)`